### PR TITLE
Fixed compiler errors on Mac OS X

### PR DESCRIPTION
**Pull request by jannschu**
Hi!

I tried to compile Gamera under Mac OS X 10.11 with the current version of Xcode and got a few errors.

Here are my attempts in fixing them. I hope this is useful :smile:

Greetings

### CODE REVIEW

File: include/knnga.hpp [hunk 1]
```diff
@@ -241,7 +241,7 @@ namespace Gamera { namespace GA {
 
             for (size_t i = 0; i < bestIndi.size(); ++i) {
                 this->knn->selection_vector[(*this->indexRelation)[i]] = bestIndi[i];
-                this->bestSolution[(*this->indexRelation)[i]] = bestIndi[i];
+                this->bestSolution[(*this->indexRelation)[i]] = (bool)bestIndi[i];
```
Comment (jannschu) [inline include/knnga.hpp]: I not sure about the correct type here 😕. `(bool)` is an educated guess…

**Comment by jannschu** [comment]
Just noticed, that there is another compiler error for the code generated for the plugins: In `gamera/plugins/_misc_filters.cpp` on line 149ff I changed it to
```cpp
return_arg = Gamera::rank(*((OneBitImageView*)self_arg), rank_arg, k_arg, border_treatment_arg);
```

I'm not sure how to change this in the code generator, as I'm new to the code at the moment.

File: include/rle_data.hpp [hunk 2]
```diff
@@ -326,7 +326,7 @@ namespace Gamera {
 	m_pos--;
 	if (!check_chunk()) {
 	  if (m_i != m_vec->m_data[m_chunk].begin()) {
-	    iterator prev_i = prev(m_i);
+	    iterator prev_i = std::prev(m_i);
```
Comment (jannschu) [inline include/rle_data.hpp]: There were convenience functions `prev` and `next` defined at the [top in this file](https://github.com/jannschu/gamera/blob/bf8e1a20c848f1412148544a2786b7dc545e472d/include/rle_data.hpp#L155), which do the same as `std::prev` and `std::next`, I think.

File: include/knnga.hpp (hunk 1 above)
Comment (cdalitz) [inline include/knnga.hpp]: Seems to be a reasonable guess, because the vector is initialized with false before.

File: include/rle_data.hpp (hunk 2 above)
Comment (cdalitz) [inline include/rle_data.hpp]: Apparently, these functions have been introduced into C++ 2011. That's why the function definition is activated only #ifdef __cplusplus < 201103. I will add the correct namespace (std) and merge your changes into the main branch. Thanks!

**Pull request merged**